clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -231,7 +230,8 @@ public boolean matches(AbstractRequest body) {
     @Test
     public void testQuotaMetrics() throws Exception {
         MockSelector selector = new MockSelector(time);
-        Sensor throttleTimeSensor = Sender.throttleTimeSensor(metrics);
+        SenderMetricsRegistry metricsRegistry = new SenderMetricsRegistry(metrics.config().tags().keySet());
```

Comment:
Can we make a `SenderMetricsRegistry metricsRegistry` a private field of the class that takes the private `metrics` object, and reuse it in all the test functions?

Reply:
Unfortunately, the answer is both yes and no. 

Yes, in this particular case (in `testQuotaMetrics()`)

In the rest of the places, no, not without touching more lines of code. The reason is, the setup() method sets up the private `metrics` object. However, most of the tests actually define _their own_ metrics object (like, see Line 237 below). And the two aren't interchangeable, because they were set up with different tags. And, similarly, the SenderMetricsRegistry() has to be set up using the same tags as the corresponding `metrics` object, so we can make the SenderMetricsRegistry() interchangeable as well.

That said, I will make the setup() method also set up a `senderMetricsRegistry` object. I will switch testQuotaMetrics() to use it and then in the future, if/when this file gets cleaned up to all use the same `metrics` object, then we will make them all use the same `senderMetricsRegistry` object.

I could switch all the tests to use the _same_ metrics object, but I didn't want to do a lot of refactoring in this PR. I wanted to keep this PR focused on adding the Sender metrics.